File: RemoteBees-iOS/RemoteBeesService/Domain/Job.swift
/***************************************************************************
 * This source file is part of the RemoteBees open source project.         *
 *                                                                         *
 * Licensed under the MIT License. See LICENSE for license information     *
 ***************************************************************************/

import Foundation

public struct Jobs: Codable {
    public let legalNotice: String
    public let jobCount: Int
    public let jobs: [Job]
    
    private enum CodingKeys: String, CodingKey {
        case legalNotice = "0-legal-notice"
        case jobCount = "job-count"
        case jobs
    }
}

public struct Job: Hashable, Codable, Identifiable {
    public let id: Int
    public let url: String
    public let title: String
    public let companyName: String
    public let category: String
    public let tags: [String]
    public let jobType: String
    public let publicationDate: Date
    public let candidateRequiredLocation: String
    public let salary: String
    public let description: String
    public let documentType: String = "JOB"

    private enum CodingKeys: String, CodingKey {
        case id
        case url
        case title
        case companyName = "company_name"
        case category
        case tags
        case jobType = "job_type"
        case publicationDate = "publication_date"
        case candidateRequiredLocation = "candidate_required_location"
        case salary
        case description
        case documentType
    }
}

File: RemoteBees-Android/RemoteBeesService/src/main/java/dev/beehive/remotebees/service/domain/Job.kt
/***************************************************************************
 * This source file is part of the RemoteBees open source project.         *
 *                                                                         *
 * Licensed under the MIT License. See LICENSE for license information     *
 ***************************************************************************/

package dev.beehive.remotebees.service.domain

import com.squareup.moshi.Json
import java.util.*

data class Jobs(
    @Json(name="0-legal-notice")val legalNotice: String,
    @Json(name="job-count") val jobCount: Int,
    val jobs: List<Job>
)

data class Job(
    val id: Int,
    val url: String,
    val title: String,
    @Json(name ="company_name") val companyName: String,
    val category: String,
    val tags: List<String>,
    @Json(name="job_type") val jobType: String,
    @Json(name="publication_date") val publicationDate: String,
    @Json(name="candidate_required_location") val candidateRequiredLocation: String,
    val salary: String,
    val description: String,
    val documentType: String = "JOB"
)
